metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -130,74 +133,132 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         ]
 
     def _get_datasets(self, catalog: Catalog) -> Iterable[Identifier]:
-        for namespace in catalog.list_namespaces():
-            yield from catalog.list_tables(namespace)
+        namespaces = catalog.list_namespaces()
+        LOGGER.debug(
+            f"Retrieved {len(namespaces)} namespaces, first 10: {namespaces[:10]}"
+        )
+        tables_count = 0
+        for namespace in namespaces:
+            tables = catalog.list_tables(namespace)
+            tables_count += len(tables)
+            LOGGER.debug(
+                f"Retrieved {len(tables)} tables for namespace: {namespace}, in total retrieved {tables_count}, first 10: {tables[:10]}"
```

Comment:
I think this should go as well to report

Reply:
Done